scipy/stats/tests/test_morestats.py

```diff
@@ -2524,3 +2524,32 @@ def test_basic(self):
         exp_stat, exp_p, dof, e = stats.chi2_contingency(tbl, correction=False)
         assert_allclose(stat, exp_stat)
         assert_allclose(p, exp_p)
+
+class TestSpherfuncs:
```

Comment:
In 2D, if you extract angles from the vectors, would this be compatible to the circular mean function? Might be a good test.

Reply:
That's a very good idea, will come back to it over the next weeks. In the meanwhile: could you maybe comment on the axis and nan decorator? I am not sure if it is applicable here. If yes, it would probably save me a lot of pain with handling edge cases ;).

Reply:
Done